### ISSUE TITLE
Error: Cannot infer key and value ...

### ISSUE THREAD

**Issue by iyegoroff** [fable-compiler/fable-react-native#40]
I'm receiving this error only on Android in [this project](https://github.com/iyegoroff/react-native-image-filter-kit/tree/master/FilterConstructor). It is triggered by `Padding (dip 123.)` line [here](https://github.com/iyegoroff/react-native-image-filter-kit/blob/664a5ee391ddc6d8c3f09339f6151ac6648b2ec3/FilterConstructor/src/Main.fs#L23). Also I can't reproduce it in a simple example, because if I comment [lines](https://github.com/iyegoroff/react-native-image-filter-kit/blob/664a5ee391ddc6d8c3f09339f6151ac6648b2ec3/FilterConstructor/src/Main.fs#L34-L35) with `CombinedFilteredImage.Message` the error will disappear.

![screenshot_20180920-013846](https://user-images.githubusercontent.com/4447438/45786813-ec392480-bc7a-11e8-893e-e50b24867175.png)

looks related https://github.com/fable-compiler/Fable/issues/1290

**Comment by alfonsogarciacaro** [comment]
Hmm, strange. I'd like to test this with Fable 2 to see if it's still happening...

**Comment by iyegoroff** [comment]
Looks like some JavaScriptCore bug led to this behavior. I didn't find any info about it, but `this.tag` in the following snippet was set to `undefined` right inside the constructor in certain situations:
```js
  setType("Fable.Helpers.ReactNative.Props.TransformsStyle", TransformsStyle);
  const FlexStyle = __exports.FlexStyle = class FlexStyle {
    constructor(tag, data) {
      this.tag = tag | 0;
      this.data = data;
    }
    ...
```
After replacing original JSC with [this one](https://github.com/react-community/jsc-android-buildscripts#how-to-use-it-with-my-react-native-app) I didn't receive this error anymore.
